### PR TITLE
Remove rounded corner from AvatarToolbarButton's highlight on animation

### PR DESCRIPTION
**Pull request by sangwoo108**
When it's animating, it expands wide. In this case, the button's radius is different from that of Highlight. In order to fill the entire button, just draw rect and let the button clip the corner.

### Bug case
image

Resolves https://github.com/brave/brave-browser/issues/25579

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Manual
* Create a new profile from app menu
* When a new window opens, wait for the avatar button to exapnd so that we can see the user name.
* When promo bubble closes, try moving mouse over to the button.
* The fill effect should fill the entire area of the button.

### CODE REVIEW

File: chromium_src/chrome/browser/ui/views/profiles/avatar_toolbar_button_delegate.cc [hunk 1]
```diff
@@ -1,35 +0,0 @@
...
-        g_browser_process->profile_manager()
-            ->GetProfileAttributesStorage()
-            .GetProfileAttributesWithPath(profile_->GetPath());
-    if (entry &&
-        entry->GetAvatarIconIndex() == profiles::GetPlaceholderAvatarIndex()) {
-      return AvatarToolbarButton::State::kNormal;
-    }
-  }
-  return state;
-}
```
Comment (sangwoo108) [inline chromium_src/chrome/browser/ui/views/profiles/avatar_toolbar_button_delegate.cc:31]: It looks dead code to me, which always returns the parent class's implementation. If we remove this, we don't need this chromium_src/ override becuase we don't need that "virtual" overriding.

File: chromium_src/chrome/browser/ui/views/profiles/avatar_toolbar_button_delegate.h [hunk 2]
```diff
@@ -1,43 +0,0 @@
...
-// want to redefine them elsewhere.
-#include "base/memory/weak_ptr.h"
-#include "build/build_config.h"
-#include "chrome/browser/profiles/profile_attributes_storage.h"
-#include "chrome/browser/ui/browser_list_observer.h"
-#include "chrome/browser/ui/views/profiles/avatar_toolbar_button.h"
-#include "components/signin/public/identity_manager/identity_manager.h"
-#include "ui/gfx/image/image.h"
-
-#define GetState virtual GetState
```
Comment (sangwoo108) [inline chromium_src/chrome/browser/ui/views/profiles/avatar_toolbar_button_delegate.h:19]: here, we needed this before.

File: browser/ui/views/profiles/brave_avatar_toolbar_button.cc [hunk 3]
```diff
@@ -38,26 +38,58 @@ namespace {
...
+
+    if (avatar_button_->GetAvatarButtonState() ==
+        AvatarToolbarButton::State::kAnimatedUserIdentity) {
+      // When it's animating we shouldn't draw rounded highlight as the button
+      // has different radius. We draw rect here then the button will clip the
+      // highlight properly.
+      path.addRect(gfx::RectToSkRect(rect));
+    } else {
+      path.addRoundRect(gfx::RectToSkRect(rect), kHighlightRadius,
+                        kHighlightRadius);
```
Comment (sangwoo108) [inline browser/ui/views/profiles/brave_avatar_toolbar_button.cc:70]: q. do we need to draw rounded rect? looks like the button can clip the corners.

**Comment by simonhong** [comment]
Highlight radius was changed.
With this PR:
image
Original:
image

**Comment by sangwoo108** [comment]
Oh, I thought it worked but it's not. I'll check out again.

**Comment by sangwoo108** [comment]
@simonhong I've changed overridden method so that it could update the radius properly. I believe this now works better.

File: browser/ui/views/profiles/brave_avatar_toolbar_button.cc [hunk 4]
```diff
@@ -38,26 +39,53 @@ namespace {
...
+  absl::optional<gfx::RRectF> GetRoundRect(const gfx::RectF& bounds) override {
+    gfx::Rect rect(avatar_button_->size());
+    rect.Inset(GetToolbarInkDropInsets(avatar_button_.get()));
+    DCHECK(avatar_button_);
+    if (avatar_button_->GetAvatarButtonState() ==
+        AvatarToolbarButton::State::kAnimatedUserIdentity) {
+      // When it's animating we shouldn't draw rounded highlight as the button
+      // has different radius. We draw rect here then the button will clip the
+      // highlight properly.
+      return gfx::RRectF(gfx::RectF(rect), 0);
```
Comment (simonhong) [inline browser/ui/views/profiles/brave_avatar_toolbar_button.cc:63]: It seems clipping doesn't work as expected like below.
Screenshot 2023-01-15 at 11 32 33 AM
I think setting same radius like below would make more better?
```
      return gfx::RRectF(gfx::RectF(rect),
                         ChromeLayoutProvider::Get()->GetCornerRadiusMetric(
                             views::Emphasis::kMaximum, {}));
```

Comment (sangwoo108) [reply browser/ui/views/profiles/brave_avatar_toolbar_button.cc:63]: Yeah, sounds good! will try.

Comment (sangwoo108) [reply browser/ui/views/profiles/brave_avatar_toolbar_button.cc]: Looks working great! Thanks.

**Review by simonhong — APPROVED** [review]
++ 👍🏼

**Review by goodov — APPROVED** [review]

**Pull request merged**